pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/AggregationFunctionUtils.java

```diff
@@ -182,4 +187,37 @@ public static String concatArgs(List<String> arguments) {
     return (arguments.size() > 1) ? String.join(CompilerConstants.AGGREGATION_FUNCTION_ARG_SEPARATOR, arguments)
         : arguments.get(0);
   }
+
+  /**
+   * Compiles and returns all transform expressions required for computing the aggregation, group-by
+   * and order-by
+   *
+   * @param brokerRequest Broker Request
+   * @param functionContexts Aggregation Function contexts
+   * @return Set of compiled expressions in the aggregation, group-by and order-by clauses
+   */
+  public static Set<TransformExpressionTree> collectExpressionsToTransform(BrokerRequest brokerRequest,
+      AggregationFunctionContext[] functionContexts) {
+
+    Set<TransformExpressionTree> expressionTrees = new LinkedHashSet<>();
+    for (AggregationFunctionContext functionContext : functionContexts) {
+      AggregationFunction function = functionContext.getAggregationFunction();
+
+      // Count is always treated as count(*) and does not need any columns to be projected.
+      if (!function.getType().equals(AggregationFunctionType.COUNT)) {
```

Comment:
No need to specialize this. Just make CountAggregationFunction return empty list

Reply:
👍